sphinxext/icon.py

```diff
@@ -1,121 +1,337 @@
 import re
-from docutils import nodes
-from docutils.parsers.rst import roles
+import fett
+from docutils import nodes, statemachine
+from docutils.parsers.rst import roles, Directive
 from sphinx.util.nodes import split_explicit_title
+from docutils.utils.error_reporting import ErrorString
 
 PAT_STRIP = re.compile(r'-(?:o|(?:alt))$')
+ALL_SUFFIXES = ('-add', '-edit', '-exclamation', '-remove', '-restart')
+HELP_TEMPLATE = fett.Template('''
+.. list-table::
 
-# A pre-approved list of icons. Approval required to add to this list.
+{{ for name in icons }}
+   * - ``{{ name }}``
+     - :{{ role }}:`{{ name }}`
+{{ end }}
+''')
+
+# Each of the below classes describes an icon font. Each must provide:
+# * A name for the docutils icon node
+# * A name for the docutils role
+# * A prefix for the CSS class attached to the output <span>
+# * A set  of icon names
+# Approval required to add to icon lists.
 # Make sure you update the documentation in github.com:mongodb/docs-meta
 # as well when updating.
-ICONS = set([
-    'caret-right',
-    'caret-square-left',
-    'caret-square-right',
-    'check-circle',
-    'check-square',
-    'copy',
-    'edit',
-    'exclamation-circle',
-    'eye',
-    'globe',
-    'lock',
-    'pencil-alt',
-    'save',
-    'star',
-    'sync-alt',
-    'trash',
-    'trash-alt',
-    'users'
-])
-
-# Discouraged, but not prohibited (yet)
-ICONS.add('arrow-right')
-ICONS.add('asterisk')
-ICONS.add('check')
-ICONS.add('ellipsis-h')
-ICONS.add('minus')
-ICONS.add('plus')
-
-ICONSB = set([
-    'windows',
-])
+
+
+class IconsFA5:
+    """FontAwesome 5 Solid icon font"""
+    NODE_NAME = 'IconFA5'
+    ROLE_NAMES = ('icon-fa5', 'icon')
+    CSS_PREFIX = 'fas fa-'
+    ICONS = set([
+        'caret-right',
+        'caret-square-left',
+        'caret-square-right',
+        'check-circle',
+        'check-square',
+        'copy',
+        'edit',
+        'exclamation-circle',
+        'eye',
+        'globe',
+        'lock',
+        'pencil-alt',
+        'save',
+        'star',
+        'sync-alt',
+        'trash',
+        'trash-alt',
+        'users'
+    ])
+
+    # Discouraged, but not prohibited (yet)
+    ICONS.add('arrow-right')
```

Comment:
Are these added separately since they're deprecated? Maybe add a single set/list instead of individual list items?
```python
DEPRECATED_ICONS = set([
    'arrow-right',
    'asterisk',
    'check',
    'ellipsis-h',
    'minus',
    'plus'
])
ICONS.update(DEPRECATED_ICONS)
```
or
```python
# Discouraged, but not prohibited (yet)
ICONS.update([
    'arrow-right',
    'asterisk',
    'check',
    'ellipsis-h',
    'minus',
    'plus'
])
```

Reply:
:+1: